synapse/push/mailer.py

```diff
@@ -364,8 +368,11 @@ def make_summary_text(self, notifs_by_room, state_by_room,
             if len(notifs_by_room[room_id]) == 1:
                 # There is just the one notification, so give some detail
                 event = notif_events[notifs_by_room[room_id][0]["event_id"]]
-                if ("m.room.member", event.sender) in state_by_room[room_id]:
-                    state_event = state_by_room[room_id][("m.room.member", event.sender)]
+                if ("m.room.member", event.sender) in room_state_ids[room_id]:
```

Comment:
It used to be a map of `room_id -> tuple -> event` and is now `room_id -> tuple -> event_id`, so the check should still make sense